include/sharg/parser.hpp

```diff
@@ -231,66 +232,53 @@ class parser
      *                        value. Must model sharg::validator.
      *
      * \param[in, out] value The variable in which to store the given command line argument.
-     * \param[in] short_id The short identifier for the option (e.g. 'a').
-     * \param[in] long_id The long identifier for the option (e.g. "age").
-     * \param[in] desc The description of the option to be shown in the help page.
-     * \param[in] spec Advanced option specification, see sharg::option_spec.
-     * \param[in] option_validator A sharg::validator that verifies the value after parsing (callable).
+     * \param[in] config A configuration object to customise the sharg::parser behaviour. See sharg::config.
+     *
+     * The `config.validator` must be applicable to the given output variable (\p value).
      *
      * \throws sharg::design_error
      */
-    template <typename option_type, validator validator_type = detail::default_validator>
+    template <typename option_type, typename validator_type>
```

Comment:
validator concept should already fail when setting validator in config?

Reply:
Jup, I added a static assert in sharg::config